### PR TITLE
feat(webtorrent): Upgrade Webtorrent to v1.9.7

### PR DESCRIPTION
**Pull request by muliswilliam**
This PR updates webtorrent library used in webtorrent extension to v1.9.7.
The very latest version v2.3.6 introduces a lot of breaking changes esp. around streaming multimedia so we settled for v1.9.7 that works with minimal changes.

Resolves https://github.com/brave/brave-browser/issues/35385

## Submitter Checklist:

- [x] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:

https://github.com/brave/brave-core/assets/48117473/99dff9b4-390b-4484-8c3a-19d143afd67d

### CODE REVIEW

**Comment by diracdeltas** [comment]
Will need a quick security review (i'm happy to do it)

**Comment by muliswilliam** [comment]
> Will need a quick security review (i'm happy to do it)

@diracdeltas Can we reuse this issue to track https://github.com/brave/reviews/issues/1524. Also please note I settled for 1.9.7 instead of v2.1.36. I have added the reasons in the description for this PR, mainly too many breaking changes that would need more time and effort to resolve.

File: components/brave_webtorrent/extension/components/torrentFileList.tsx [hunk 1]
```diff
@@ -80,64 +81,96 @@ const tableHeader: Cell[] = [
...
-    return <Container>
+  return (
+    <Container>
       <HeaderRow>
         <Header>Files</Header>
         <Button
           onClick={saveAllFiles}
           disabled={torrent?.progress !== 1}
-          text="Save all files"
+          text='Save all files'
```
Comment (josheleonard) [inline components/brave_webtorrent/extension/components/torrentFileList.tsx:156]: Should this use a localization string?

File: components/brave_webtorrent/extension/components/torrentFileList.tsx [hunk 2]
```diff
@@ -80,64 +81,96 @@ const tableHeader: Cell[] = [
...
+          header={tableHeader}
+          rows={rows}
+        >
+          {torrent && !torrent.files ? (
+            <LoadingContainer>
+              <Spinner />
+              Loading the torrent file list
+            </LoadingContainer>
+          ) : (
+            'Click "Start Torrent" to begin your download.'
```
Comment (josheleonard) [inline components/brave_webtorrent/extension/components/torrentFileList.tsx:170]: localization?

File: components/brave_webtorrent/extension/components/torrentFileList.tsx [hunk 3]
```diff
@@ -80,64 +81,96 @@ const tableHeader: Cell[] = [
...
+                <span>{file.name}</span>
+              )
           },
           {
-            content: torrent.serverURL && <SaveButton text="Save file"
-              level="secondary"
-              onClick={() => downloadFile(`${torrent.serverURL}/${index}/${file.name}`, file.name)}/>
+            content: torrent.serverURL && (
+              <SaveButton
+                text='Save file'
```
Comment (josheleonard) [inline components/brave_webtorrent/extension/components/torrentFileList.tsx:123]: localization?

File: components/brave_webtorrent/extension/components/mediaViewer.tsx [hunk 4]
```diff
@@ -49,41 +56,84 @@ interface Props {
...
-        {fileType === 'iframe' && <iframe id='iframe' src={fileURL} sandbox='allow-same-origin' />}
-        {fileType === 'unknown' && <div>Unsupported file type</div>}
-      </>}
-  </Container>
+  return (
+    <Container>
+      {loading ? (
+        <LoadingContainer>
+          <Spinner />
+          Loading media...
```
Comment (josheleonard) [inline components/brave_webtorrent/extension/components/mediaViewer.tsx:94]: localization?

File: components/brave_webtorrent/extension/components/mediaViewer.tsx [hunk 5]
```diff
@@ -49,41 +56,84 @@ interface Props {
...
+            />
+          )}
+          {fileType === 'iframe' && (
+            <iframe
+              id='iframe'
+              src={fileURL}
+              sandbox='allow-same-origin'
+            />
+          )}
+          {fileType === 'unknown' && <div>Unsupported file type</div>}
```
Comment (josheleonard) [inline components/brave_webtorrent/extension/components/mediaViewer.tsx:134]: localization?

File: components/brave_webtorrent/extension/brave_webtorrent.tsx [hunk 6]
```diff
@@ -23,8 +24,10 @@
 // to brave_webtorrent2.html and have that rewrite the URL to
 // brave_webtorrent.html
 if (window.location.pathname === '/extension/brave_webtorrent2.html') {
-  window.location.href =
-    window.location.href.replace('brave_webtorrent2.html', 'brave_webtorrent.html')
+  window.location.href = window.location.href.replace(
```
[the remark this answers is not in the document]
Comment (muliswilliam) [reply components/brave_webtorrent/extension/brave_webtorrent.tsx]: @SocketSecurity ignore npm/http-node@1.2.0

**Comment by diracdeltas** [comment]
@SocketSecurity ignore npm/http-node@1.2.0

**Review by josheleonard — APPROVED** [review]
Frontend ++

**Review by diracdeltas — APPROVED** [review]

**Pull request merged**